contracts/Strategy.sol

```diff
@@ -16,32 +16,106 @@ import {
     IERC20,
     Address
 } from "@openzeppelin/contracts/token/ERC20/SafeERC20.sol";
+import "@openzeppelin/contracts/math/Math.sol";
 
-// Import interfaces for many popular DeFi projects, or add your own!
-//import "../interfaces/<protocol>/<Interface>.sol";
+import "./interfaces/UniswapInterfaces/IUniswapV2Router02.sol";
+import {IVault as IComplifiVault} from "./interfaces/complifi/IVault.sol";
+import {ILiquidityMining} from "./interfaces/complifi/ILiquidityMining.sol";
 
 contract Strategy is BaseStrategy {
     using SafeERC20 for IERC20;
     using Address for address;
     using SafeMath for uint256;
 
+    IComplifiVault public tokenVault =
+        IComplifiVault(address(0xea5b9650f6c47D112Bb008132a86388B594Eb849));
+
+    ILiquidityMining public constant liquidityMining =
+        ILiquidityMining(address(0x8a5827Ad1f28d3f397B748CE89895e437b8ef90D));
+
+    IERC20 public constant comfi =
+        IERC20(address(0x752Efadc0a7E05ad1BCCcDA22c141D01a75EF1e4));
+
+    address private constant router =
+        address(0x7a250d5630B4cF539739dF2C5dAcb4c659F2488D);
+    address private constant weth =
+        address(0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2);
+
+    address[] public path;
+
     constructor(address _vault) public BaseStrategy(_vault) {
-        // You can set these parameters on deployment to whatever you want
-        // maxReportDelay = 6300;
-        // profitFactor = 100;
-        // debtThreshold = 0;
+        want.safeApprove(address(tokenVault), type(uint256).max);
+        primaryToken().safeApprove(address(tokenVault), type(uint256).max);
```

Comment:
when you use a method more than once, assign to a local var.

Reply:
do you mean `address(tokenVault)`?

Reply:
```address primaryToken = primaryToken();

primaryToken.firstUsage();
primaryToken.secondUsage();```